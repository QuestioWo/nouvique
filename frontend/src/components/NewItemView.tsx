import React from 'react';

import { Result } from 'neverthrow';
import { Col, Button, Form, Row } from 'react-bootstrap';
import { RouteComponentProps } from 'react-router';

import { ItemREST, ItemSubmitREST, itemSubmitRESTLink, PageProps, resolvePOSTCall, ItemTypeRESTNewSubmit, ItemTypeREST, itemTypeRESTNewSubmitLink } from '../utils';

import BasePage from './elements/BasePage';

import './NewItemView.css';

export interface NewItemViewProps extends RouteComponentProps, PageProps { }

interface State extends ItemSubmitREST, ItemTypeRESTNewSubmit {
	name: string
}

export class NewItemView extends React.Component<NewItemViewProps, State> {
	constructor(props: NewItemViewProps) {
		super(props);

		this.state = {
			bio: "",
			shape: "",
			colour: "",
			tag0: "",
			tag1: "",
			tag2: "",
			tag3: "",
			tag4: "",
			pictures: [],
			name: "",
			price: 0,
			quantity: 0,
			size: "",
			available: true
		};
	}

	handleImageUpdate = (event: any) => {
		var pictures: string[] = [];
		for (var i = 0; i < event.target.files.length; ++i) {
			const reader = new FileReader();
			reader.onload = () => {
				var settingString;
				if (reader.result === null || reader.result instanceof ArrayBuffer) {
					settingString = "";
				} else {
					settingString = reader.result;
				}
				pictures.push(settingString);
			};

			reader.readAsDataURL(event.target.files[i]);
		}

		this.setState({ pictures: pictures });
	}

	handleFormSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
		event.preventDefault();

		const data: ItemSubmitREST = this.state;

		const path = itemSubmitRESTLink + localStorage.username + "/" + this.state.name + "/";

		const result: Result<ItemREST, Error> = await resolvePOSTCall<ItemREST, ItemSubmitREST>(path, data, true);

		result
			.map(async res => {
				const dataType: ItemTypeRESTNewSubmit = this.state;

				const path = itemTypeRESTNewSubmitLink + localStorage.username + "/" + this.state.name + "/";

				const result: Result<ItemTypeREST, Error> = await resolvePOSTCall<ItemTypeREST, ItemTypeRESTNewSubmit>(path, dataType, true);

				result
					.map(resType => {
						this.props.updateAlertBar("Successfully created item", "success", true);

						this.props.history.push('/item/' + localStorage.username + "/" + res.name);

						return null; // necessary to silence warning
					})
					.mapErr(errType => {
						const message: string = "Item already exists with that name, try again with a new name or re-log in";

						this.props.updateAlertBar(message, "danger", true);

						console.error(errType);
					});

				return null; // necessary to silence warning
			})
			.mapErr(err => {
				const message: string = "Item already exists with that name, try again with a new name or re-log in";

				this.props.updateAlertBar(message, "danger", true);

				console.error(err);
			});
	}

	render() {
		return (
			<React.Fragment>
				<BasePage {...this.props}>
					<Row>
						<h2 className="title">
							create a new item
						</h2>
					</Row>

					<Form onSubmit={this.handleFormSubmit}>
						<Form.Group className="mb-3">
							<Form.Label>name</Form.Label>
							<Form.Control
								required
								type="text"
								value={this.state.name}
								onChange={(event: any) => { this.setState({ name: event.target.value }) }} />
						</Form.Group>

						<Form.Group className="mb-3">
							<Form.Label>bio</Form.Label>
							<Form.Control
								required
								type="text"
								as="textarea"
								value={this.state.bio}
								placeholder='description'
								onChange={(event: any) => { this.setState({ bio: event.target.value }) }} />
						</Form.Group>

						<Form.Group className="mb-3">
							<Form.Label>shape/style</Form.Label>
							<Form.Control
								required
								type="text"
								value={this.state.shape}
								placeholder='shirt'
								onChange={(event: any) => { this.setState({ shape: event.target.value }) }} />
						</Form.Group>

						<Form.Group className="mb-3">
							<Form.Label>colour</Form.Label>
							<Form.Control
								required
								type="text"
								value={this.state.colour}
								placeholder='black'
								onChange={(event: any) => { this.setState({ colour: event.target.value }) }} />
						</Form.Group>

						<Row>
							<Form.Label>tags</Form.Label>
						</Row>
						<Row>
							<Col>
								<Form.Group className="mb-3">
									<Form.Control
										type="text"
										value={this.state.tag0}
										placeholder='modern'
										onChange={(event: any) => { this.setState({ tag0: event.target.value }) }} />
								</Form.Group>
							</Col>
							<Col>
								<Form.Group className="mb-3">
									<Form.Control
										type="text"
										value={this.state.tag1}
										placeholder='streetwear'
										onChange={(event: any) => { this.setState({ tag1: event.target.value }) }} />
								</Form.Group>
							</Col>
							<Col>
								<Form.Group className="mb-3">
									<Form.Control
										type="text"
										value={this.state.tag2}
										placeholder='trendy'
										onChange={(event: any) => { this.setState({ tag2: event.target.value }) }} />
								</Form.Group>
							</Col>
							<Col>
								<Form.Group className="mb-3">
									<Form.Control
										type="text"
										value={this.state.tag3}
										placeholder='kitsch'
										onChange={(event: any) => { this.setState({ tag3: event.target.value }) }} />
								</Form.Group>
							</Col>
							<Col>
								<Form.Group className="mb-3">
									<Form.Control
										type="text"
										value={this.state.tag4}
										placeholder='classy'
										onChange={(event: any) => { this.setState({ tag4: event.target.value }) }} />
								</Form.Group>
							</Col>
						</Row>

						<Form.Group className="mb-3">
							<Form.Label>price</Form.Label>
							<Form.Control
								required
								type="number"
								value={this.state.price}
								onChange={(event: any) => { this.setState({ price: event.target.value }) }} />
						</Form.Group>

						<Form.Group className="mb-3">
							<Form.Label>size</Form.Label>
							<Form.Control
								required
								type="text"
								value={this.state.size}
								placeholder='Medium'
								onChange={(event: any) => { this.setState({ size: event.target.value }) }} />
						</Form.Group>

						<Form.Group className="mb-3">
							<Form.Label>quantity</Form.Label>
							<Form.Control
								required
								type="number"
								value={this.state.quantity}
								onChange={(event: any) => { this.setState({ quantity: event.target.value }) }} />
						</Form.Group>

						<Form.Group className="mb-3" controlId="profile_picture">
							<Form.Label>pictures (.png, 2Mb max, 5 images max)</Form.Label>
							<Form.Control
								required
								type="file"
								accept={'.png'}
								multiple
								onChange={this.handleImageUpdate} />
						</Form.Group>

						<Button type='submit' className='styled-btn' variant='outline-success'>
							submit
						</Button>
					</Form>
				</BasePage>
			</React.Fragment>
		);
	}
}