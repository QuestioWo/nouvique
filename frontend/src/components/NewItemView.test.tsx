import * as React from 'react';

import { mount, shallow } from 'enzyme';

import { NewItemView, NewItemViewProps } from './NewItemView';
import { MemoryRouter as Router } from 'react-router';
import { loadStripe } from '@stripe/stripe-js';

describe('NewItemView', () => {
	const getProps = (): NewItemViewProps => (
		{
			history: {} as any,
			location: {} as any,
			match: { params: { item_id: "foo" } as any } as any,
			updateAlertBar: async () => { },
			emptyBasket: () => { },
			addToBasket: () => { },
			removeFromBasket: () => { },
			getBasketItems: () => [],
			getTotalBasketPrice: () => 0,
			checkBasketAvailabilities: async () => { },
			stripePromise: loadStripe("foo"),
			alert: {} as any
		}
	)

	const getComponent = (props: NewItemViewProps) => (
		<NewItemView {...props} />
	);


	it('Should render correctly with shallow', () => {
		const component = shallow(
			getComponent(getProps())
		);
		expect(component.debug()).toMatchSnapshot();
	});

	it('Should render correctly with mount', () => {
		// have to include router as will fail with react-router problem due to use of specific functions
		const component = mount(
			<Router>
				{getComponent(getProps())}
			</Router>
		);
		expect(component.debug()).toMatchSnapshot();
	});
});