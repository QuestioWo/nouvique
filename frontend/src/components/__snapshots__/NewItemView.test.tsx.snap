// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewItemView Should render correctly with mount 1`] = `
"<MemoryRouter>
  <Router history={{...}}>
    <NewItemView history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
      <BasePage history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
        <div className=\\"master\\">
          <img alt=\\"\\" src=\\"/logo_main.png\\" className=\\"logo\\" />
           
          <NavBar>
            <Navbar bg=\\"transparent\\" variant=\\"light\\" expand={true} collapseOnSelect={false}>
              <nav className=\\"navbar navbar-expand navbar-light bg-transparent\\">
                <NavbarCollapse className=\\"justify-content-center\\">
                  <ForwardRef in={false} className=\\"justify-content-center\\" timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} getDimensionValue={[Function: getDefaultDimensionValue]}>
                    <ForwardRef addEndListener={[Function: transitionEndListener]} in={false} timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} aria-expanded={{...}} onEnter={[Function (anonymous)]} onEntering={[Function (anonymous)]} onEntered={[Function (anonymous)]} onExit={[Function (anonymous)]} onExiting={[Function (anonymous)]} childRef={{...}}>
                      <Transition in={false} timeout={300} mountOnEnter={false} unmountOnExit={false} appear={false} aria-expanded={{...}} onEnter={[Function (anonymous)]} onEntered={[Function (anonymous)]} onEntering={[Function (anonymous)]} onExit={[Function (anonymous)]} onExited={[Function (anonymous)]} onExiting={[Function (anonymous)]} addEndListener={[Function (anonymous)]} nodeRef={{...}} enter={true} exit={true}>
                        <div className=\\"justify-content-center navbar-collapse collapse\\" aria-expanded={{...}}>
                          <Nav className=\\"mr-auto nav_style\\" justify={false} fill={false}>
                            <ForwardRef as=\\"div\\" activeKey={[undefined]} className=\\"mr-auto nav_style navbar-nav\\" onSelect={[Function (anonymous)]}>
                              <div className=\\"mr-auto nav_style navbar-nav\\" onKeyDown={[Function: handleKeyDown]} role={[undefined]}>
                                <NavItem className=\\"item_style\\">
                                  <div className=\\"item_style nav-item\\">
                                    <NavLink className=\\"nav-link\\" isActive={[Function: isActive]} activeClassName=\\"nav-link active\\" to={{...}}>
                                      <Link aria-current=\\"page\\" className=\\"nav-link nav-link active\\" style={{...}} to={{...}}>
                                        <LinkAnchor aria-current=\\"page\\" className=\\"nav-link nav-link active\\" style={{...}} href=\\"/\\" navigate={[Function: navigate]}>
                                          <a aria-current=\\"page\\" className=\\"nav-link nav-link active\\" style={{...}} href=\\"/\\" onClick={[Function: onClick]}>
                                            Home
                                          </a>
                                        </LinkAnchor>
                                      </Link>
                                    </NavLink>
                                  </div>
                                </NavItem>
                                <NavItem className=\\"item_style\\">
                                  <div className=\\"item_style nav-item\\">
                                    <NavLink className=\\"nav-link\\" activeClassName=\\"nav-link active\\" to={{...}}>
                                      <Link aria-current={{...}} className=\\"nav-link\\" style={[undefined]} to={{...}}>
                                        <LinkAnchor aria-current={{...}} className=\\"nav-link\\" style={[undefined]} href=\\"/browse\\" navigate={[Function: navigate]}>
                                          <a aria-current={{...}} className=\\"nav-link\\" style={[undefined]} href=\\"/browse\\" onClick={[Function: onClick]}>
                                            Browse
                                          </a>
                                        </LinkAnchor>
                                      </Link>
                                    </NavLink>
                                  </div>
                                </NavItem>
                                <NavItem className=\\"item_style\\">
                                  <div className=\\"item_style nav-item\\">
                                    <NavLink className=\\"nav-link\\" activeClassName=\\"nav-link active\\" to={{...}}>
                                      <Link aria-current={{...}} className=\\"nav-link\\" style={[undefined]} to={{...}}>
                                        <LinkAnchor aria-current={{...}} className=\\"nav-link\\" style={[undefined]} href=\\"/ethics\\" navigate={[Function: navigate]}>
                                          <a aria-current={{...}} className=\\"nav-link\\" style={[undefined]} href=\\"/ethics\\" onClick={[Function: onClick]}>
                                            Ethics
                                          </a>
                                        </LinkAnchor>
                                      </Link>
                                    </NavLink>
                                  </div>
                                </NavItem>
                                <NavItem className=\\"item_style\\">
                                  <div className=\\"item_style nav-item\\">
                                    <NavLink className=\\"nav-link nav-right-side\\" activeClassName=\\"nav-link active nav-right-side\\" to={{...}}>
                                      <Link aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} to={{...}}>
                                        <LinkAnchor aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/login\\" navigate={[Function: navigate]}>
                                          <a aria-current={{...}} className=\\"nav-link nav-right-side\\" style={[undefined]} href=\\"/login\\" onClick={[Function: onClick]}>
                                            Login
                                          </a>
                                        </LinkAnchor>
                                      </Link>
                                    </NavLink>
                                  </div>
                                </NavItem>
                              </div>
                            </ForwardRef>
                          </Nav>
                        </div>
                      </Transition>
                    </ForwardRef>
                  </ForwardRef>
                </NavbarCollapse>
              </nav>
            </Navbar>
          </NavBar>
          <div className=\\"mainbody\\">
            <AlertBar history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
              <Alert show={true} variant={[undefined]} onClose={[Function (anonymous)]} dismissible={true} transition={{...}} closeLabel=\\"Close alert\\">
                <Fade unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false}>
                  <ForwardRef addEndListener={[Function: transitionEndListener]} unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false} onEnter={[Function (anonymous)]} childRef={{...}}>
                    <Transition unmountOnExit={true} in={true} timeout={300} mountOnEnter={false} appear={false} onEnter={[Function (anonymous)]} onEntered={[Function (anonymous)]} onEntering={[Function (anonymous)]} onExit={[Function (anonymous)]} onExited={[Function (anonymous)]} onExiting={[Function (anonymous)]} addEndListener={[Function (anonymous)]} nodeRef={{...}} enter={true} exit={true}>
                      <div role=\\"alert\\" className=\\"fade alert alert-dismissible show\\">
                        <CloseButton onClick={[Function (anonymous)]} aria-label=\\"Close alert\\" variant={[undefined]}>
                          <button type=\\"button\\" className=\\"btn-close\\" onClick={[Function (anonymous)]} aria-label=\\"Close alert\\" />
                        </CloseButton>
                      </div>
                    </Transition>
                  </ForwardRef>
                </Fade>
              </Alert>
            </AlertBar>
            <Row>
              <div className=\\"row\\">
                <h2 className=\\"title\\">
                  create a new item
                </h2>
              </div>
            </Row>
            <Form onSubmit={[Function (anonymous)]}>
              <form onSubmit={[Function (anonymous)]} className=\\"\\">
                <FormGroup className=\\"mb-3\\">
                  <div className=\\"mb-3\\">
                    <FormLabel column={false} visuallyHidden={false}>
                      <label className=\\"form-label\\" htmlFor={[undefined]}>
                        name
                      </label>
                    </FormLabel>
                    <FormControl required={true} type=\\"text\\" value=\\"\\" onChange={[Function: onChange]}>
                      <input required={true} value=\\"\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                    </FormControl>
                  </div>
                </FormGroup>
                <FormGroup className=\\"mb-3\\">
                  <div className=\\"mb-3\\">
                    <FormLabel column={false} visuallyHidden={false}>
                      <label className=\\"form-label\\" htmlFor={[undefined]}>
                        bio
                      </label>
                    </FormLabel>
                    <FormControl required={true} type=\\"text\\" as=\\"textarea\\" value=\\"\\" placeholder=\\"description\\" onChange={[Function: onChange]}>
                      <textarea required={true} value=\\"\\" placeholder=\\"description\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                    </FormControl>
                  </div>
                </FormGroup>
                <FormGroup className=\\"mb-3\\">
                  <div className=\\"mb-3\\">
                    <FormLabel column={false} visuallyHidden={false}>
                      <label className=\\"form-label\\" htmlFor={[undefined]}>
                        shape/style
                      </label>
                    </FormLabel>
                    <FormControl required={true} type=\\"text\\" value=\\"\\" placeholder=\\"shirt\\" onChange={[Function: onChange]}>
                      <input required={true} value=\\"\\" placeholder=\\"shirt\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                    </FormControl>
                  </div>
                </FormGroup>
                <FormGroup className=\\"mb-3\\">
                  <div className=\\"mb-3\\">
                    <FormLabel column={false} visuallyHidden={false}>
                      <label className=\\"form-label\\" htmlFor={[undefined]}>
                        colour
                      </label>
                    </FormLabel>
                    <FormControl required={true} type=\\"text\\" value=\\"\\" placeholder=\\"black\\" onChange={[Function: onChange]}>
                      <input required={true} value=\\"\\" placeholder=\\"black\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                    </FormControl>
                  </div>
                </FormGroup>
                <Row>
                  <div className=\\"row\\">
                    <FormLabel column={false} visuallyHidden={false}>
                      <label className=\\"form-label\\" htmlFor={[undefined]}>
                        tags
                      </label>
                    </FormLabel>
                  </div>
                </Row>
                <Row>
                  <div className=\\"row\\">
                    <Col>
                      <div className=\\"col\\">
                        <FormGroup className=\\"mb-3\\">
                          <div className=\\"mb-3\\">
                            <FormControl type=\\"text\\" value=\\"\\" placeholder=\\"modern\\" onChange={[Function: onChange]}>
                              <input value=\\"\\" placeholder=\\"modern\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                            </FormControl>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                    <Col>
                      <div className=\\"col\\">
                        <FormGroup className=\\"mb-3\\">
                          <div className=\\"mb-3\\">
                            <FormControl type=\\"text\\" value=\\"\\" placeholder=\\"streetwear\\" onChange={[Function: onChange]}>
                              <input value=\\"\\" placeholder=\\"streetwear\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                            </FormControl>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                    <Col>
                      <div className=\\"col\\">
                        <FormGroup className=\\"mb-3\\">
                          <div className=\\"mb-3\\">
                            <FormControl type=\\"text\\" value=\\"\\" placeholder=\\"trendy\\" onChange={[Function: onChange]}>
                              <input value=\\"\\" placeholder=\\"trendy\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                            </FormControl>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                    <Col>
                      <div className=\\"col\\">
                        <FormGroup className=\\"mb-3\\">
                          <div className=\\"mb-3\\">
                            <FormControl type=\\"text\\" value=\\"\\" placeholder=\\"kitsch\\" onChange={[Function: onChange]}>
                              <input value=\\"\\" placeholder=\\"kitsch\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                            </FormControl>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                    <Col>
                      <div className=\\"col\\">
                        <FormGroup className=\\"mb-3\\">
                          <div className=\\"mb-3\\">
                            <FormControl type=\\"text\\" value=\\"\\" placeholder=\\"classy\\" onChange={[Function: onChange]}>
                              <input value=\\"\\" placeholder=\\"classy\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                            </FormControl>
                          </div>
                        </FormGroup>
                      </div>
                    </Col>
                  </div>
                </Row>
                <FormGroup className=\\"mb-3\\">
                  <div className=\\"mb-3\\">
                    <FormLabel column={false} visuallyHidden={false}>
                      <label className=\\"form-label\\" htmlFor={[undefined]}>
                        price
                      </label>
                    </FormLabel>
                    <FormControl required={true} type=\\"number\\" value={0} onChange={[Function: onChange]}>
                      <input required={true} value={0} onChange={[Function: onChange]} type=\\"number\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                    </FormControl>
                  </div>
                </FormGroup>
                <FormGroup className=\\"mb-3\\">
                  <div className=\\"mb-3\\">
                    <FormLabel column={false} visuallyHidden={false}>
                      <label className=\\"form-label\\" htmlFor={[undefined]}>
                        size
                      </label>
                    </FormLabel>
                    <FormControl required={true} type=\\"text\\" value=\\"\\" placeholder=\\"Medium\\" onChange={[Function: onChange]}>
                      <input required={true} value=\\"\\" placeholder=\\"Medium\\" onChange={[Function: onChange]} type=\\"text\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                    </FormControl>
                  </div>
                </FormGroup>
                <FormGroup className=\\"mb-3\\">
                  <div className=\\"mb-3\\">
                    <FormLabel column={false} visuallyHidden={false}>
                      <label className=\\"form-label\\" htmlFor={[undefined]}>
                        quantity
                      </label>
                    </FormLabel>
                    <FormControl required={true} type=\\"number\\" value={0} onChange={[Function: onChange]}>
                      <input required={true} value={0} onChange={[Function: onChange]} type=\\"number\\" size={[undefined]} readOnly={[undefined]} id={[undefined]} className=\\"form-control\\" />
                    </FormControl>
                  </div>
                </FormGroup>
                <FormGroup className=\\"mb-3\\" controlId=\\"profile_picture\\">
                  <div className=\\"mb-3\\">
                    <FormLabel column={false} visuallyHidden={false}>
                      <label className=\\"form-label\\" htmlFor=\\"profile_picture\\">
                        pictures (.png, 2Mb max, 5 images max)
                      </label>
                    </FormLabel>
                    <FormControl required={true} type=\\"file\\" accept=\\".png\\" multiple={true} onChange={[Function (anonymous)]}>
                      <input required={true} accept=\\".png\\" multiple={true} onChange={[Function (anonymous)]} type=\\"file\\" size={[undefined]} readOnly={[undefined]} id=\\"profile_picture\\" className=\\"form-control\\" />
                    </FormControl>
                  </div>
                </FormGroup>
                <Button type=\\"submit\\" className=\\"styled-btn\\" variant=\\"outline-success\\" active={false} disabled={false}>
                  <button disabled={false} type=\\"submit\\" className=\\"styled-btn btn btn-outline-success\\">
                    submit
                  </button>
                </Button>
              </form>
            </Form>
          </div>
        </div>
        <footer className=\\"footer py-3 text-black\\">
          <div />
          <Container fluid={false}>
            <div className=\\"container\\">
              <Row>
                <div className=\\"row\\">
                  <Col>
                    <div className=\\"col\\">
                      Nouvique 2021 v
                      0.22.0
                    </div>
                  </Col>
                  <Col>
                    <div className=\\"col\\">
                      <div style={{...}}>
                        <Link to=\\"/support\\">
                          <LinkAnchor href=\\"/support\\" navigate={[Function: navigate]}>
                            <a href=\\"/support\\" onClick={[Function: onClick]}>
                              Support
                            </a>
                          </LinkAnchor>
                        </Link>
                      </div>
                    </div>
                  </Col>
                </div>
              </Row>
            </div>
          </Container>
        </footer>
      </BasePage>
    </NewItemView>
  </Router>
</MemoryRouter>"
`;

exports[`NewItemView Should render correctly with shallow 1`] = `
"<Fragment>
  <BasePage history={{...}} location={{...}} match={{...}} updateAlertBar={[Function: updateAlertBar]} emptyBasket={[Function: emptyBasket]} addToBasket={[Function: addToBasket]} removeFromBasket={[Function: removeFromBasket]} getBasketItems={[Function: getBasketItems]} getTotalBasketPrice={[Function: getTotalBasketPrice]} checkBasketAvailabilities={[Function: checkBasketAvailabilities]} stripePromise={{...}} alert={{...}}>
    <Row>
      <h2 className=\\"title\\">
        create a new item
      </h2>
    </Row>
    <Form onSubmit={[Function (anonymous)]}>
      <FormGroup className=\\"mb-3\\">
        <FormLabel column={false} visuallyHidden={false}>
          name
        </FormLabel>
        <FormControl required={true} type=\\"text\\" value=\\"\\" onChange={[Function: onChange]} />
      </FormGroup>
      <FormGroup className=\\"mb-3\\">
        <FormLabel column={false} visuallyHidden={false}>
          bio
        </FormLabel>
        <FormControl required={true} type=\\"text\\" as=\\"textarea\\" value=\\"\\" placeholder=\\"description\\" onChange={[Function: onChange]} />
      </FormGroup>
      <FormGroup className=\\"mb-3\\">
        <FormLabel column={false} visuallyHidden={false}>
          shape/style
        </FormLabel>
        <FormControl required={true} type=\\"text\\" value=\\"\\" placeholder=\\"shirt\\" onChange={[Function: onChange]} />
      </FormGroup>
      <FormGroup className=\\"mb-3\\">
        <FormLabel column={false} visuallyHidden={false}>
          colour
        </FormLabel>
        <FormControl required={true} type=\\"text\\" value=\\"\\" placeholder=\\"black\\" onChange={[Function: onChange]} />
      </FormGroup>
      <Row>
        <FormLabel column={false} visuallyHidden={false}>
          tags
        </FormLabel>
      </Row>
      <Row>
        <Col>
          <FormGroup className=\\"mb-3\\">
            <FormControl type=\\"text\\" value=\\"\\" placeholder=\\"modern\\" onChange={[Function: onChange]} />
          </FormGroup>
        </Col>
        <Col>
          <FormGroup className=\\"mb-3\\">
            <FormControl type=\\"text\\" value=\\"\\" placeholder=\\"streetwear\\" onChange={[Function: onChange]} />
          </FormGroup>
        </Col>
        <Col>
          <FormGroup className=\\"mb-3\\">
            <FormControl type=\\"text\\" value=\\"\\" placeholder=\\"trendy\\" onChange={[Function: onChange]} />
          </FormGroup>
        </Col>
        <Col>
          <FormGroup className=\\"mb-3\\">
            <FormControl type=\\"text\\" value=\\"\\" placeholder=\\"kitsch\\" onChange={[Function: onChange]} />
          </FormGroup>
        </Col>
        <Col>
          <FormGroup className=\\"mb-3\\">
            <FormControl type=\\"text\\" value=\\"\\" placeholder=\\"classy\\" onChange={[Function: onChange]} />
          </FormGroup>
        </Col>
      </Row>
      <FormGroup className=\\"mb-3\\">
        <FormLabel column={false} visuallyHidden={false}>
          price
        </FormLabel>
        <FormControl required={true} type=\\"number\\" value={0} onChange={[Function: onChange]} />
      </FormGroup>
      <FormGroup className=\\"mb-3\\">
        <FormLabel column={false} visuallyHidden={false}>
          size
        </FormLabel>
        <FormControl required={true} type=\\"text\\" value=\\"\\" placeholder=\\"Medium\\" onChange={[Function: onChange]} />
      </FormGroup>
      <FormGroup className=\\"mb-3\\">
        <FormLabel column={false} visuallyHidden={false}>
          quantity
        </FormLabel>
        <FormControl required={true} type=\\"number\\" value={0} onChange={[Function: onChange]} />
      </FormGroup>
      <FormGroup className=\\"mb-3\\" controlId=\\"profile_picture\\">
        <FormLabel column={false} visuallyHidden={false}>
          pictures (.png, 2Mb max, 5 images max)
        </FormLabel>
        <FormControl required={true} type=\\"file\\" accept=\\".png\\" multiple={true} onChange={[Function (anonymous)]} />
      </FormGroup>
      <Button type=\\"submit\\" className=\\"styled-btn\\" variant=\\"outline-success\\" active={false} disabled={false}>
        submit
      </Button>
    </Form>
  </BasePage>
</Fragment>"
`;
